Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ListController.cs

```diff
@@ -74,7 +74,14 @@ public ActionResult ByCategory(int? id)
 
             if (id.HasValue && this.contestCategoriesData.HasContestsById(id.Value))
             {
-                contestCategory.Contests = this.GetContestsInCategory(id.Value);
+                page = page ?? 1;
+
+                contestCategory.Contests = this.contestsData
```

Comment:
Why don't you encapsulate the whole query in the data service?

Reply:
I think separation of concerns is good enough reason to extract logic, even if it likely won't see any re-usability at all. However I agree that the data layer should not know about `ViewModels`, because then we are coupling our business logic with the presentation layer.